### ISSUE TITLE
Shipment Schedules not created for new flatrate terms

### ISSUE THREAD

**Issue by metasnw** [metasfresh/metasfresh#2678]
### Is this a bug or feature request?
bug

### What is the current behavior?
exception on asynch work package

#### Which are the steps to reproduce?
1. have a condition with transistion of type subscription and weekly delivery
1. create a new order
1. OK: contract is created
1. OK: subscription progress is created
1. NOK: no shipment schedules

sample order:  https://w101.metasfresh.com:8443/window/143/1000076
condition: https://w101.metasfresh.com:8443/window/540120/1000005
wp error:
https://w101.metasfresh.com:8443/window/363/3632143

log in work package:

```
org.adempiere.exceptions.AdempiereException: NullPointerException
	at org.adempiere.exceptions.AdempiereException.wrapIfNeeded(AdempiereException.java:86)
	at org.adempiere.exceptions.AdempiereException.wrapIfNeeded(AdempiereException.java:82)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.appendAndLogHowtoDisableMessage(AnnotatedModelInterceptor.java:573)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.executeNow(AnnotatedModelInterceptor.java:563)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.execute(AnnotatedModelInterceptor.java:509)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.execute(AnnotatedModelInterceptor.java:463)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.onModelChange(AnnotatedModelInterceptor.java:441)
	at org.adempiere.ad.modelvalidator.ModelInterceptor2ModelValidatorWrapper.modelChange(ModelInterceptor2ModelValidatorWrapper.java:123)
	at org.compiere.model.ModelValidationEngine.invokeModelChangeMethod(ModelValidationEngine.java:943)
	at org.compiere.model.ModelValidationEngine.invokeModelChangeMethods(ModelValidationEngine.java:920)
	at org.compiere.model.ModelValidationEngine.fireModelChange0(ModelValidationEngine.java:802)
	at org.compiere.model.ModelValidationEngine.access$000(ModelValidationEngine.java:96)
	at org.compiere.model.ModelValidationEngine$1.run(ModelValidationEngine.java:755)
	at org.compiere.model.ModelValidationEngine$2.run(ModelValidationEngine.java:779)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$3.call(TrxCallableWrappers.java:145)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$3.call(TrxCallableWrappers.java:135)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call0(AbstractTrxManager.java:771)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call(AbstractTrxManager.java:683)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.run(AbstractTrxManager.java:591)
	at org.compiere.model.ModelValidationEngine.executeInTrx(ModelValidationEngine.java:774)
	at org.compiere.model.ModelValidationEngine.fireModelChange(ModelValidationEngine.java:750)
	at org.compiere.model.PO.fireModelChange(PO.java:4885)
	at org.compiere.model.PO.save0(PO.java:2860)
	at org.compiere.model.PO.access$100(PO.java:137)
	at org.compiere.model.PO$1.run(PO.java:2767)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$3.call(TrxCallableWrappers.java:145)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$3.call(TrxCallableWrappers.java:135)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call0(AbstractTrxManager.java:771)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call(AbstractTrxManager.java:683)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call(AbstractTrxManager.java:584)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call(AbstractTrxManager.java:515)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.run(AbstractTrxManager.java:500)
	at org.compiere.model.PO.saveEx(PO.java:2760)
	at org.adempiere.model.POWrapper.save(POWrapper.java:787)
	at org.adempiere.model.InterfaceWrapperHelper.save(InterfaceWrapperHelper.java:541)
	at de.metas.contracts.inoutcandidate.SubscriptionShipmentScheduleHandler.createCandidatesFor(SubscriptionShipmentScheduleHandler.java:122)
	at de.metas.inoutcandidate.api.impl.ShipmentScheduleHandlerBL.createMissingCandidates(ShipmentScheduleHandlerBL.java:163)
	at de.metas.inoutcandidate.async.CreateMissingShipmentSchedulesWorkpackageProcessor.processWorkPackage(CreateMissingShipmentSchedulesWorkpackageProcessor.java:80)
	at de.metas.async.processor.impl.WorkpackageProcessorTask.processWorkpackage(WorkpackageProcessorTask.java:284)
	at de.metas.async.processor.impl.WorkpackageProcessorTask.access$000(WorkpackageProcessorTask.java:84)
	at de.metas.async.processor.impl.WorkpackageProcessorTask$1.run(WorkpackageProcessorTask.java:167)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$2.call(TrxCallableWrappers.java:103)
	at org.adempiere.ad.trx.api.impl.TrxCallableWrappers$2.call(TrxCallableWrappers.java:93)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call0(AbstractTrxManager.java:771)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.call(AbstractTrxManager.java:683)
	at org.adempiere.ad.trx.api.impl.AbstractTrxManager.run(AbstractTrxManager.java:591)
	at de.metas.async.processor.impl.WorkpackageProcessorTask.run(WorkpackageProcessorTask.java:157)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at Thread.run(Thread.java:748)
Caused by: NullPointerException
	at de.metas.handlingunits.shipmentschedule.api.impl.HUShipmentScheduleBL.createEffectiveValues(HUShipmentScheduleBL.java:799)
	at de.metas.handlingunits.model.validator.M_ShipmentSchedule.createEffectiveValues(M_ShipmentSchedule.java:111)
	at sun.reflect.GeneratedMethodAccessor189.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at reflect.Method.invoke(Method.java:498)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.executeNow0(AnnotatedModelInterceptor.java:607)
	at org.adempiere.ad.modelvalidator.AnnotatedModelInterceptor.executeNow(AnnotatedModelInterceptor.java:559)
	... 48 more

```

**Comment by metas-ts** [comment]
CPed to the release branch

**Comment by metas-rc** [comment]
Results of IT

I Created the testing data by inspiring from https://github.com/metasfresh/metasfresh/issues/2502.
The differences I did were: 
* Transition: Make Subscription interval one week
* Sys Config: I modified it to be 15 so I have more entries in shipment schedules

* I created the order for another partner so it doesn't overlap with my already exsting term.
* I selected the flatate conditions that I defined for the weekly delivery
* I completed the order
* => OK: My Contract was created
* =>OK:  SubscriptionProgress entries were created
* =>OK:  Shipment schedules were created( for the timerange covered by my number of days defined in the sys config).
